suiteData.js

```diff
@@ -12,8 +15,14 @@ class SuiteData {
             throw new Error("A suite 'name' parameter should be included on the query string or in the request body.");
         }
         var tests = obj && obj.tests;
+        if (_.isString(tests) && fs.existsSync(sanitize(tests))) {
```

Comment:
also, it's not expected that a constructor does complicated, long, async stuff. that's why you probably used the sync versions of the fs functions.
you'll need to decide where this logic fits best, in the init or in the fromRequest methods...